connaisseur/tests/integration/integration-test.sh

```diff
@@ -51,6 +51,51 @@ else
   echo 'Successfully allowed usage of signed image'
 fi
 
+echo 'Testing edge case of tag defined in both targets and release json file...'
```

Comment:
This test case means essentially two digests for one tag, correct? Can that only happen with these two files?

Reply:
Tag can be e.g. signed by different people, but it will not show up in the current content. But if you push stuff before setting up delegations you can end up with multiplte "current" digests for a tag. See e.g. `docker trust inspect securesystemsengineering/testimage --pretty` and you'll see that there are two different digests for `signed`. For `double_sig` there's only one shown even though both exist (as is tested by the integration test introduced in one of the commits). @phbelitz is better at explaining how and why that happens. We just noticed again that it did happen and I decided this would be a good time to introduce the test that Connaisseur actually chooses the correct image in that case

Reply:
@annekebr yea the explaination goes like this:

let's say you don't have delegations set up. then your signed digest will pop up in the `targets.json`, as it should. after you signed some images this way you come up with the great idea to introduce delegations. sure thing. now if you sign your images with the newly introduced delegation keys, your signed digests will pop up in the `targets/releases.json`, under the same tag which is already stored in the `targets.json`. now you have two different digests (an old one in the `targets.json` and a new one in the `targets/releases.json`) for the same tag. this test essentially should make sure the "right" one is used, so the new one in the `targets/releases.json`, since this is now the source of truth with delegations set up.

funnily enough from here there is yet another edge case! so when you already signed images without delegations, meaning their digest pop up in the `targets.json`, and then introduce delegation WITHOUT signing with them yet, you get a funny situation. since delegations are set up, you'd expect the "right" digest to be in the `targets/releases.json`, like in the case from above, but nope! you haven't used the delegation keys yet to sign any digests, so only the `targets.json` will actually contain any digests.